### ISSUE TITLE
DRB3 and DRB5

### ISSUE THREAD

**Issue by bsb2014** [deepomicslab/SpecHLA#14]
Hi, can SpecHLA type DRB3 and DRB5 as kourami? Thanks

**Comment by wshuai294** [comment]
Hello, now SpecHLA does not support DRB3 and DRB5.

**Comment by bsb2014** [comment]
Do you have any plans to make SpecHLA support additional HLAs?

**Comment by bsb2014** [comment]
How does the SpecHLA phase the variants when only short reads are used? Thanks

**Comment by wshuai294** [comment]
Yes. Our group is developing another package to support additional HLA genes. I will let you know once it is finished.

**Comment by wshuai294** [comment]
If the short reads cannot afford enough phase information, SpecHLA will use HLA database, genotype frequency, and pedigree information to phase the variants.

**Comment by bsb2014** [comment]
It would be great if the next release could type HLA using diploid assembly. Thanks

**Comment by wshuai294** [comment]
Great suggestion. We do have some code to extract the sequence of HLA genes from the diploid assembly. I will update SpecHLA to support typing from diploid assembly as soon as possible.

**Comment by wshuai294** [comment]
Hello,
Thanks for the valuable suggestion. As suggested, we added a module to support HLA typing from diploid assemblies (`typing_from_assembly.py`). Please see the changes in SpecHLA v1.0.3 or the latest commit.

**Comment by lidd77** [comment]
> Yes. Our group is developing another package to support additional HLA genes. I will let you know once it is finished.

Hello, shuai,
       Are there any other tools  can work for long read HLA typing ( nanopore sequencing)  ? 
       if we just add DRB3/4/5 imgt sequence into  SpecHLA db ,  SpeckHLA can detect accurately  DRB3/4/5 allele  ?

Expecting your reply ! 
thanks .

**Comment by wshuai294** [comment]
Hello,
Sorry for the delayed update. If you add DRB3/4/5 imgt sequence into SpecHLA db, it still needs some revisions in the scripts. I believe it will work well after some adjustments. I wonder do you have any benchmark data (with real data and ground truth)? With real benchmark data, we can add this function quickly. 

Also, as I know HLA*LA also can work with nanopore sequencing data. But it might require too many computational resources. You can have a try on it.

Best,
Shuai

**Comment by lidd77** [comment]
> Hello, Sorry for the delayed update. If you add DRB3/4/5 imgt sequence into SpecHLA db, it still needs some revisions in the scripts. I believe it will work well after some adjustments. I wonder do you have any benchmark data (with real data and ground truth)? With real benchmark data, we can add this function quickly.
> 
> Also, as I know HLA*LA also can work with nanopore sequencing data. But it might require too many computational resources. You can have a try on it.
> 
> Best, Shuai

Yeah,  I have  got   the benchmark data for A,B,C ,DRB1, DQA1,DQB1, DPA1, DPB1,  DRB3/4/5 ,  that's why I hope  you nice guy can  support DRB3/4/5 as soon as possible .   
could you update specHLA to support DRB3/4/5 this week ?
Thank you very much !!!

**Comment by wshuai294** [comment]
So great! We will update it very soon. This week is a little challenging but I believe we can finish it next week. Could you please send us the benchmark data, so that we can start to adjust the parameters.

**Comment by lidd77** [comment]
> So great! We will update it very soon. This week is a little challenging but I believe we can finish it next week. Could you please send us the benchmark data, so that we can start to adjust the parameters.

well,  what is your email address  ?

**Comment by wshuai294** [comment]
Thanks. My email is swang66-c@my.cityu.edu.hk

**Comment by alisamatisse** [comment]
Hello again Shuai,

So is there an update to know if analyzed sequence contains DRB-3/4/5 and type it? 

Thanks!

**Comment by wshuai294** [comment]
Hello, we plan to release the update to support DRB3/4/5 within one month.

**Comment by LeeDarian** [comment]
> Hello, we plan to release the update to support DRB3/4/5 within one month.

look forward to

**Comment by wshuai294** [comment]
We're excited to share that SpecHLA has been updated and rebranded as SpecImmune, now supporting a much broader range of immune-related genes. Explore it here: [SpecImmune on GitHub](https://github.com/deepomicslab/SpecImmune).

**Comment by zplizzi** [comment]
makes sense, thanks!
